core/renderers/common/info.js

```diff
@@ -266,7 +267,7 @@ Blockly.blockRendering.RenderInfo.prototype.populateTopRow_ = function() {
         new Blockly.blockRendering.RoundCorner(this.constants_));
   }
 
-  if (hasHat) {
+  if (hasHat && !hasOutputConnection) {
```

Comment:
I would fold the !hasOutputConnection into the definition of hasHat.

Reply:
There's actually more going on here. We should also consider whether we're connected to a block. eg, this is what you get: 

<img width="340" alt="Screen Shot 2019-10-17 at 9 22 26 AM" src="https://user-images.githubusercontent.com/16690124/67028127-ab028680-f0bf-11e9-81a9-61937953d9f2.png">

Reply:
no prev, no output, and start hat is set?

Reply:
Fixed. Since we're checking for start hat in multiple places, thoughts on moving that check onto the block? ``block.hasHat()``?

<img width="299" alt="Screen Shot 2019-10-17 at 9 41 32 AM" src="https://user-images.githubusercontent.com/16690124/67029527-56143f80-f0c2-11e9-91d9-38410c85d625.png">

Reply:
Actually if we want renderers to be able to change this behaviour we probably shouldn't.

Reply:
This is still wrong--the top repeat block should not have a hat.  The check is whether there's a previous connection, not whether there's a previous block.

Reply:
Yep, misunderstanding on my part. Updated to only show when there's no previous connection and no output connection.